integration/eth2network/rotation_writer.go

```diff
@@ -0,0 +1,122 @@
+package eth2network
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+)
+
+type RotatingLogWriter struct {
+	dirPath         string
+	fileStartName   string
+	maxFileSize     int64
+	maxNumFiles     int
+	currentFile     *os.File
+	currentSize     int64
+	numFilesRotated int
+}
+
+func NewRotatingLogWriter(dirPath, fileStartName string, maxFileSize int64, maxNumFiles int) (*RotatingLogWriter, error) {
+	writer := &RotatingLogWriter{
+		dirPath:       dirPath,
+		fileStartName: fileStartName,
+		maxFileSize:   maxFileSize,
+		maxNumFiles:   maxNumFiles,
+	}
+
+	// Open the current log file
+	err := writer.openNextFile()
+	if err != nil {
+		return nil, err
+	}
+
+	return writer, nil
+}
+
+func (w *RotatingLogWriter) openNextFile() error {
+	if w.currentFile != nil {
+		err := w.currentFile.Close()
+		if err != nil {
+			fmt.Println(err)
+		}
+	}
+
+	w.currentSize = 0
+	w.numFilesRotated = 0
+
+	// Construct the file name for the next log file
+	timestamp := time.Now().Format("2006-01-02T15-04-05")
```

Comment:
My ignorance of go - but is that the correct format specifier?

Reply:
Yeah it is a weird format approach, I'm not really sure of the story behind it, but it is.